ui/components/component-library/select-wrapper/select-wrapper.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import type { PolymorphicRef, BoxProps } from '../box';
+import { Box, Popover, PopoverPosition } from '..';
+import {
+  SelectWrapperComponent,
+  SelectWrapperProps,
+} from './select-wrapper.types';
+import { SelectContext } from './select-wrapper.context';
+
+export const SelectWrapper: SelectWrapperComponent = React.forwardRef(
+  <C extends React.ElementType = 'div'>(
+    {
+      className = '',
+      defaultValue,
+      value,
+      onValueChange,
+      placeholder,
+      isDanger,
+      isDisabled,
+      isOpen,
+      onOpenChange,
+      isMultiSelect, // Prevents the uncontrolled open state from being toggled
+      triggerComponent,
+      popoverProps,
+      children,
+      onBlur, // Controlled onBlur prop
+      ...props
+    }: SelectWrapperProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    const [uncontrolledValue, setUncontrolledValue] = useState<any | null>();
+    const [isUncontrolledOpen, setIsUncontrolledOpen] =
+      useState<boolean>(false);
+    const [referenceElement, setReferenceElement] =
+      useState<HTMLElement | null>();
+
+    const setBoxRef = (popoverRef: HTMLElement | null) => {
+      setReferenceElement(popoverRef);
+    };
+
+    const toggleUncontrolledOpen = () => {
+      setIsUncontrolledOpen(!isUncontrolledOpen);
+    };
+
+    const handleBlur = (e: any) => {
+      const wrapper = e.currentTarget;
+      const { relatedTarget } = e;
+
+      if (!wrapper.contains(relatedTarget)) {
```

Comment:
If relatedTarget is `null` this will throw an error. It might be worth adding a check for this? 
```suggestion
      if (relatedTarget && !wrapper.contains(relatedTarget)) {
```

Reply:
Also do we need to also add the popover ref and make sure that it doesn't close if just the menu is clicked?

Take the Radix as an example


https://github.com/MetaMask/metamask-extension/assets/8112138/fef29891-7a2a-4d35-9626-1a433c2536d5

vs

https://github.com/MetaMask/metamask-extension/assets/8112138/47d02f20-4eb5-4151-bcd1-91a5ace3c869

Reply:
I did not. I was assuming if the triggerComponent wasn't a focusable element then it could cause an issue but that might mean that onBlur wouldn't even trigger so it might be fine.

Reply:
@brianacnguyen and I were discussing something unrelated and we happened upon this issue. We believe the `Popover` should have 2 additional props: `onClose` and `isClosedOnOutsideClick`.  There is logic in the ModalContent component as an example https://github.com/MetaMask/metamask-extension/blob/82ceffb6d699b9deacb7caed74e6a9919aabf012/ui/components/component-library/modal-content/modal-content.tsx#L50-L69

And then the `onBlur` logic can be passed to the `onClose` prop handled by the`Popover`

```
<Popover onClose={onBlur} />
```